src/runtime/virtcontainers/qemu_amd64.go

```diff
@@ -191,3 +206,54 @@ func (q *qemuAmd64) appendImage(ctx context.Context, devices []govmmQemu.Device,
 func (q *qemuAmd64) appendBridges(devices []govmmQemu.Device) []govmmQemu.Device {
 	return genericAppendBridges(devices, q.Bridges, q.qemuMachine.Type)
 }
+
+// enable protection
+func (q *qemuAmd64) enableProtection() error {
+	flags, err := CPUFlags(procCPUInfo)
+	if err != nil {
+		return err
+	}
+
+	// TDX is supported and properly loaded when the firmware directory exists or `tdx` is part of the CPU flags
```

Comment:
I trust you on this, but I'm surprised the condition is not "and" instead of "or".

How do you use `tdx` if the flag is set but the firmware is not present?

Reply:
this depends on the CPU version, it can have both or only one